birdhouse/read-configs.include.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/sh
+# This script is meant to be sourced by all scripts needing to read default.env
+# or env.local.
+#
+# Normally only read_configs() is needed, but for any special needs, each step
+# can be called individually.
+#
+# When not using read_configs(), caller is responsible for knowing the proper
+# sequence to read the config files, for hiding password when reading env.local
+# and to call process_delayed_eval() to have fully usable resolved variable
+# values.
+
+# Derive COMPOSE_DIR from the most probable locations.
+# This is NOT meant to be exhautive.
+# Caller of this file can simply set COMPOSE_DIR itself.
+discover_compose_dir() {
+    if [ -z "$COMPOSE_DIR" -o ! -e "$COMPOSE_DIR" ]; then
+        if [ -e "./pavics-compose.sh" ]; then
+            # Current dir is COMPOSE_DIR
+            COMPOSE_DIR="`realpath .`"
+        elif [ -e "../pavics-compose.sh" ]; then
+            # Parent dir is COMPOSE_DIR
+            # Case of all the scripts under deployment/ or scripts/
+            COMPOSE_DIR="`realpath ..`"
+        elif [ -e "../birdhouse-deploy/birdhouse/pavics-compose.sh" ]; then
+            # Case of sibling checkout at same level as birdhouse-deploy.
+            COMPOSE_DIR="`realpath "../birdhouse-deploy/birdhouse"`"
+        elif [ -e "../../birdhouse-deploy/birdhouse/pavics-compose.sh" ]; then
+            # Case of subdir of sibling checkout at same level as birdhouse-deploy.
+            COMPOSE_DIR="`realpath "../../birdhouse-deploy/birdhouse"`"
+        elif [ -e "../../../birdhouse-deploy/birdhouse/pavics-compose.sh" ]; then
+            # Case of sub-subdir of sibling checkout at same level as birdhouse-deploy.
+            COMPOSE_DIR="`realpath "../../../birdhouse-deploy/birdhouse"`"
+        elif [ -e "./birdhouse/pavics-compose.sh" ]; then
+            # Child dir is COMPOSE_DIR
+            COMPOSE_DIR="`realpath birdhouse`"
+        fi
+        export COMPOSE_DIR
```

Comment:
Do we want to add `echo ${COMPOSE_DIR}`? 
This is a good way to pass the function "output" (`DIR=$(discover_compose_dir)`) without passing by an external variable that could be modified at any time.